src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/NativeRuntimeEventSource.Threading.NativeSinks.cs

```diff
@@ -355,12 +355,14 @@ public unsafe void ThreadPoolIOPack(NativeOverlapped* nativeOverlapped)
             LogWaitHandleWaitStart(WaitSource, AssociatedObjectID, ClrInstanceID);
         }
 
+#pragma warning disable CS8500
         [NonEvent]
         [MethodImpl(MethodImplOptions.NoInlining)]
         public unsafe void WaitHandleWaitStart(
             WaitHandleWaitSourceMap waitSource = WaitHandleWaitSourceMap.Unknown,
             object? associatedObject = null) =>
-            WaitHandleWaitStart(waitSource, *(nint*)Unsafe.AsPointer(ref associatedObject));
+            WaitHandleWaitStart(waitSource, *(nint*)&associatedObject);
```

Comment:
Would `Unsafe.As<object, nint>(ref associatedObject)` maybe look cleaner in cases like this?

Reply:
EventSource uses these unpinned object pointers in number of places. 

It may be nice to create an internal helper for this on EventSource and switch all places to call it. `internal static long ObjectIDForEvents(object o)`

Reply:
> EventSource uses these unpinned object pointers in number of places.

This isn't an unpinned pointer though but effectively a `BitCast<object, nint>`.

Reply:
NativeRuntimeEventSource.Threading.cs, Lock.cs and TplEventSource.cs are the other files that can use this helper.

For Lock.cs, I would delete the property and just call the internal helper where the event is emitted.